specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -169,75 +167,79 @@ export function TableUniquenessRules({
             >
               {commonText.save()}
             </Submit.Save>
-          )}
-        </>
-      }
-      header={header}
-      headerButtons={
-        <>
-          <span className="-ml-2 flex-1" />
+          </>
+        }
+        header={header}
+        headerButtons={
+          <>
+            <span className="-ml-2 flex-1" />
+            <Button.Small
+              aria-pressed={isEditing}
+              className="w-fit"
+              disabled={
+                !hasPermission('/schemaconfig/uniquenessrules', 'update')
+              }
+              onClick={toggleIsEditing}
+            >
+              {commonText.edit()}
+            </Button.Small>
+          </>
+        }
+        modal={false}
+        onClose={handleClose}
+      >
+        <table className="grid-table grid-cols-[2fr_1fr] gap-2 gap-y-4 overflow-auto">
+          <thead>
+            <tr>
+              <td>{schemaText.uniqueFields()}</td>
+              <td>{schemaText.scope()}</td>
+            </tr>
+          </thead>
+          {tableRules?.map(({ rule, duplicates }, index) => (
+            <UniquenessRuleRow
+              fetchedDuplicates={duplicates}
+              fields={fields}
+              isEditing={isEditing}
+              key={index}
+              label={getUniqueInvalidReason(
+                rule.scopes.map(
+                  (scope) =>
+                    getFieldsFromPath(model, scope).at(-1) as Relationship
+                ),
+                filterArray(rule.fields.map((field) => model.getField(field)))
+              )}
```

Comment:
can this be moved into UniquenessRuleRow?
there doesn't appear any variable in here that is not available inside the component as a prop

and validating validation message for a given row seems like a job of UniquenessRuleRow
(plus `label` prop name is not making it clear that the label contains a validation message)